src/ucs/memory/rcache.c

```diff
@@ -313,6 +360,13 @@ static void ucs_mem_region_destroy_internal(ucs_rcache_t *rcache,
         ucs_free(ucs_rcache_region_pfn_ptr(region));
     }
 
+    ucs_spin_lock(&rcache->lru.lock);
+    ucs_rcache_region_lru_remove(rcache, region);
+    ucs_spin_unlock(&rcache->lru.lock);
+
+    --rcache->num_regions;
```

Comment:
shouldn't it be under spin lock as well?

Reply:
AFAIU, no need for the LRU lock here. @yosefe ?